marshmallow_recipe/fields.py

```diff
@@ -834,10 +860,50 @@ def _deserialize(self, value: Any, attr: Any, data: Any, **_: Any) -> Any:
     StrField = StrFieldV2
 
     class DateTimeFieldV2(m.fields.DateTime):
+        @staticmethod
+        def __isoformat(v: datetime.datetime, *, localtime: bool = False, **kwargs: Any) -> str:
+            assert not localtime
+            assert not kwargs
+            return datetime.datetime.isoformat(v)
+
+        DATEFORMAT_SERIALIZATION_FUNCS = {
+            **m.fields.DateTime.DATEFORMAT_SERIALIZATION_FUNCS,  # type: ignore
+            **(
+                {
+                    "iso": __isoformat,
+                    "iso8601": __isoformat,
+                }
+                if sys.version_info >= (3, 11)
+                else {}
+            ),
+        }
+        DATEFORMAT_DESERIALIZATION_FUNCS = {
+            **m.fields.DateTime.DATEFORMAT_DESERIALIZATION_FUNCS,  # type: ignore
+            **(
+                {
+                    "iso": datetime.datetime.fromisoformat,
+                    "iso8601": datetime.datetime.fromisoformat,
+                }
+                if sys.version_info >= (3, 11)
+                else {}
+            ),
+        }
+
+        def _serialize(self, value: Any, attr: Any, obj: Any, **kwargs: Any) -> Any:
+            if value is None:
+                return None
+
+            if value.tzinfo is None:
```

Comment:
missing part for marshmallow 2: date-utils was used and it ultimately auto converted datetime to utc if no timezone is specified.